api/src/controllers/team.js

```diff
@@ -187,42 +185,76 @@ function updateTeamMember(req, res, next) {
-    headers: { Accept: 'application/json' },
-  });
-  console.log(response.data);
+/**
+ * Handles a GET request on /:team_id/gh_auth/:code to exchange a GitHub
+ * temporary code acquired during the first step of the GitHub OAuth flow with a
+ * GitHub access token.
+ *
+ * @see:
+ * https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps
+ *
+ * @param req request object
+ * @param res  response object
+ * @param next handler to the next middleware
+ * @returns 200 if the code was successfully exchanged for an access token
+ * @returns 400 if the exchange was unsuccessful
+ */
+async function getGHAccessToken(req, res, next) {
+  const { code } = req.params;
 
-  if (response.data.error) {
-    return res.status(400).json(response.data);
+  const { data } = await axios.post(
+    'https://github.com/login/oauth/access_token', null, {
+      headers: {
+        'Accept': 'application/json',
+      },
+      params: {
+        client_id: githubClientId,
+        client_secret: githubClientSecret,
+        code,
+      },
+    });
+  if (data.error) {
+    return next(fillErrorObject(400,
+      'Failed to exchange GitHub temporary code for GitHub Access Token.',
+      [data]));
   }
-
-  return res.status(200).json(response.data);
+  return res.status(200).json(data);
 }
 
+/**
+ * Handles a POST request on /:team_id/deploy to deploy a client's website
+ * by delegating to the Scholly service.
+ *
+ * @see:
+ * https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps
+ *
+ * @param req request object, containing in the body the GitHub token from the
+ *     OAuth flow, and the necessary data needed by Scholly
+ * @param res response object
+ * @param next handler to the next middleware
+ * @returns 200 if the deployment was successful
+ * @returns 400 if the GitHub username could not be fetched using the token
+ * @returns 500 if Scholly was unable to deploy the website
+ */
 async function deployToGHPages(req, res, next) {
-  const teamId = req.params.team_id;
+  const { team_id: teamId } = req.params;
   const { ghToken, teamPublications, teamInfo, teamMembers } = req.body;
 
-  // call github API to get username
-  const response = await axios.get('https://api.github.com/user', {
+  // Call github API to get username.
+  const { data } = await axios.get('https://api.github.com/user', {
     headers: { Authorization: 'token ' + ghToken },
   });
-
-  if (response.data.errors) {
+  if (data.errors) {
     return next(
-      fillErrorObject(400, 'Validation error', [response.data.errors[0].detail])
+      fillErrorObject(400, 'Validation error', [data.errors[0].detail])
     );
   }
 
-  const ghUsername = response.data.login;
-  console.log(ghUsername);
+  const ghUsername = data.login;
+  logger.info(`GitHub deploy initiated for user: ${ghUsername}`);
 
+  // TODO: ideally this data should be fetched by us, and we should not expect
+  //  the client to provide it.
```

Comment:
I think it's best that the data that needs to be sent to Scholly should be fetched by the `API` - atleast that was how I originally designed it. This is a separation of concerns argument and also the fact that it'll make the entire [CUJ](https://www.reforge.com/brief/critical-user-journeys-how-google-product-teams-react-when-growth-slows#2SsLPAALDeKxpwQPdEp6Uw) when they deploy the app slower since if they have 100 publications all of that is sent over the wire, and unnecessarily. I prefer the old design where the API is responsible for fetching all this info to feed into scholly.

Reply:
Do you mean you intended scholly to be fetching the data, and not the Researchify backend which passes it onto scholly?

Reply:
I think he means the Reserachify backend should not be requiring the frontend to pass this information:
```
const { ghToken, teamPublications, teamInfo, teamMembers } = req.body;
```
Which I agree, the backend can fetch that information itself. Currently the frontend queries backend for these data and then passes it backend again, so no point of having that circle. 

Maybe @aksh0001, if I'm understanding correctly, then can we place these 2 lines of comments above line 241. That would make it easier to understand.

Reply:
Yep that’s what I’m trying to convey. It doesn’t make much sense for the client to fetch this data from the backend to then send it back.
I think of all the refactors this should be a priority.

@kdav108 will do

Reply:
Done.